packages/codec/src/error.ts

```diff
@@ -0,0 +1,54 @@
+export const CODEC_OPTIONAL_PATH = "______?______";
+type CodecOptionalPath = typeof CODEC_OPTIONAL_PATH;
+export class CodecBaseParseError extends Error {
+  constructor(message: string, public expectedType: string) {
+    super(message);
+  }
+}
+
+const CODEC_EXECUTE_ERROR_NAME = "CodecExecuteError";
+export function isCodecExecuteError(
+  error: unknown
+): error is CodecExecuteError {
+  return (error as Error)?.name === CODEC_EXECUTE_ERROR_NAME;
```

Comment:
```suggestion
  if (!(error instanceof Error)) return false;
  return error.name === CODEC_EXECUTE_ERROR_NAME;
```

we should avoid using `as` if possible

Reply:
I agree. Although at here I use optional chaining for preventing access nullable object.